src/engines/html5/media-source/adapters/native-adapter.js

```diff
@@ -102,22 +106,190 @@ export default class NativeAdapter implements IMediaSourceAdapter {
   }
 
   /**
+   * @constructor
+   * @param {HTMLVideoElement} videoElement - The video element which bind to NativeAdapter
+   * @param {string} source - The source URL
+   * @param {Object} config - The media source adapter configuration
+   */
+  constructor(videoElement: HTMLVideoElement, source: Object, config: Object) {
+    this._config = config;
+    this._videoElement = videoElement;
+    this._source = source.url;
+    this._eventManager = new EventManager();
+  }
+
+  /**
    * Load the video source
    * @function load
-   * @returns {void}
+   * @returns {Promise<Object>} - The loaded data
    */
-  load(): void {
-    NativeAdapter._logger.debug('load');
-    this._videoElement.load();
+  load(): Promise<Object> {
+    if (!this._loadPromise) {
+      this._loadPromise = new Promise((resolve, reject) => {
+        this._eventManager.listen(this._videoElement, PlayerEvents.LOADED_METADATA, () => {
+          this._eventManager.unlisten(this._videoElement, PlayerEvents.LOADED_METADATA);
+          let data = {tracks: this._getParsedTracks()};
+          NativeAdapter._logger.debug('load');
+          resolve(data);
+        });
+        this._eventManager.listen(this._videoElement, PlayerEvents.ERROR, (error) => {
+          this._eventManager.unlisten(this._videoElement, PlayerEvents.ERROR);
+          NativeAdapter._logger.error(error);
+          reject(error);
+        });
+        this._videoElement.src = this._source;
+      });
+    }
+    return this._loadPromise;
   }
 
   /**
-   * Clearing the video source
+   * Clear the video source
    * @function destroy
    * @returns {void}
    */
   destroy(): void {
     NativeAdapter._logger.debug('destroy');
-    this._videoElement.src = "";
+    this._eventManager.destroy();
+    this._loadPromise = null;
```

Comment:
I'm not sure about recreating Html5. but on change media the engine should call the adapter's destroy and recreate adapter according the mime type.